pcsx2/gui/MainFrame.cpp

```diff
@@ -584,19 +587,66 @@ void MainEmuFrame::OnActivate( wxActivateEvent& evt )
 }
 // ----------------------------------------------------------------------------
 
+static void GetMenuItemLabel(MenuIdentifiers Id, wxString* label)
```

Comment:
Well, I think less time is used to determine what label and help text is used in:
```
if (wxMenuItem *cdvd_full = menubar.FindItem(fullboot_id)) {
	switch (Source) {
	case CDVD_SourceType::Iso:
		cdvd_full->SetItemLabel("Boot ISO (f&ull)");
		cdvd_full->SetHelp(_("Boot the VM using the current ISO source media"));
		break;
	case CDVD_SourceType::Plugin:
		cdvd_full->SetItemLabel("Boot CDVD (f&ull)");
		cdvd_full->SetHelp(_("Boot the VM using the current CD/DVD source media"));
		break;
	case CDVD_SourceType::NoDisc:
		cdvd_full->SetItemLabel("Boot BIOS");
		cdvd_full->SetHelp(_("Boot the VM without any source media"));
		break;
	default:
		pxAssert(false);
	}
}
```
than in:
```
if (wxMenuItem *cdvd_full = menubar.FindItem(fullboot_id))
{
	wxString label = GetMenuItemLabel(fullboot_id);
	std::map<CDVD_SourceType, wxString> help_text;
	help_text.insert(std::make_pair(CDVD_SourceType::Iso, _("Boot the VM using the current ISO source media")));
	help_text.insert(std::make_pair(CDVD_SourceType::Plugin, _("Boot the VM using the current CD/DVD source media")));
	help_text.insert(std::make_pair(CDVD_SourceType::NoDisc, _("Boot the VM without any source media")));

	cdvd_full->SetItemLabel(label);
	cdvd_full->SetHelp(help_text.at(Source));
}
```
where the help text is more or less immediately known, but you have then navigate to SetMenuItemLabel to determine what label is being used:

```
if (Id == MenuId_Boot_CDVD)
{
	switch (Source)
	{
	case CDVD_SourceType::Iso:    return _("Boot ISO (f&ull)");
	case CDVD_SourceType::Plugin: return _("Boot CDVD (f&ull)");
	case CDVD_SourceType::NoDisc: return _("Boot BIOS");
	default:
		pxAssert(false);
	}
}
```
YMMV